packages/compiler/lib/decorators.ts

```diff
@@ -813,3 +813,62 @@ export function isDeprecated(program: Program, type: Type): boolean {
 export function getDeprecated(program: Program, type: Type): string | undefined {
   return program.stateMap(deprecatedKey).get(type);
 }
+
+const overloadedByKey = Symbol("overloadedByKey");
+const overloadsOperationKey = Symbol("overloadsOperation");
+
+/**
+ * `@overload` - Indicate that the target overloads (specializes) the overloads type.
+ * @param context DecoratorContext
+ * @param target The specializing operation declaration
+ * @param overloads The operation to be overloaded.
+ */
+export function $overload(
+  context: DecoratorContext,
+  target: OperationType,
+  overloads: OperationType
+) {
+  // TODO: Verify that the target is a subtype of the overloads instance (e.g. doesn't expand on the set
+  // of parameters or return types for the overloads function)
```

Comment:
I'd be really curious how easy this is with `checker.isTypeAssignableTo`. Unless it's especially hard (which I hope it isn't), then I worry about leaving this TODO tool long and then having to make a breaking change.

Reply:
I just didn't know which method to use :). I have no problems other than TLBV (time left before vacation) to make this change before merging.

Reply:
Works pretty well!  Added usage of `isTypeAssignableTo` and a test to verify.